src/quaternion_ops.jl

```diff
@@ -0,0 +1,187 @@
+#############################################################
+#
+#  add extra quaternion operations for Quaternions
+#  some should probably be merged back into Quaternions
+#
+#############################################################
+
+
+@doc """
+    Equivilent arbitrary axis rotation angle for a quaternion
+""" ->
+rot_angle(q::Quaternion) = 2 * acos(Quaternions.normalize(q).s)
+
+
+@doc """
+function to find the mean of a set of quaternians
+
+Args:
+
+    Qarray         - Vector of quaternions
+
+    method         - if 0 (default), the returned quaternion minimizes sum_i=1:n (||R(Qi) - R(Qbar)||)  with || the frobius norm, or
+                                     equivalently minimizes  sum_i=1:n (sin(phi_i / 2))^2, where phi_i is the angle of Qe_i, with  Qbar = Q_i x Qe_i
+
+                   - if 1, the returned quaternian minimizes sum_i=1:n (phi_i ^2) where phi_i is the angle of Qe_i, with  Qbar = Q_i x Qe_i
+                           this should be about the same as method == 0 if phi_i are all small
+""" ->
+function mean(Qarray::Vector{Quaternion}, method::Integer = 0)
```

Comment:
It can be - it comes up in the prediction step of Bayesian tracking algorithms (finding the sample mean of a bunch of non-linear dynamic model transformed samples).  The dynamics in the pole top tracking are a made up at present...